cppcache/include/geode/DataOutput.hpp

```diff
@@ -497,8 +497,12 @@ class APACHE_GEODE_EXPORT DataOutput {
   static void acquireLock();
   static void releaseLock();
 
+  struct FreeDeleter {
+    void operator()(uint8_t* p) { free(p); }
+  };
+
   // memory m_buffer to encode to.
-  std::unique_ptr<uint8_t[]> m_bytes;
+  std::unique_ptr<uint8_t, FreeDeleter> m_bytes;
```

Comment:
What would it look like to convert this to a `std::vector`? Then we get all this for free.

Reply:
Not sure if this is possible. When the DataOutput constructor is invoked the m_bytes variable is assigned a static buffer which seems to be reused later when other DataOutput instances are created.